rust/lance-index/src/scalar/inverted.rs

```diff
@@ -60,19 +61,15 @@ impl InvertedIndex {
         token_ids
             .into_iter()
             .filter_map(|token| self.invert_list.retrieve(token))
-            .for_each(|(row_ids, freq)| {
+            .for_each(|row_freq| {
                 // TODO: this can be optimized by parallelizing the calculation
-                row_ids
-                    .iter()
-                    .zip(freq.iter())
-                    .for_each(|(&row_id, &freq)| {
-                        let freq = freq as f32;
-                        let bm25 = bm25.entry(row_id).or_insert(0.0);
-                        *bm25 += self.idf(row_ids.len()) * freq * (K1 + 1.0)
-                            / (freq
-                                + K1 * (1.0 - B
-                                    + B * self.docs.num_tokens[row_id as usize] as f32 / avgdl));
-                    });
+                row_freq.iter().for_each(|(row_id, freq)| {
+                    let row_id = *row_id;
+                    let freq = *freq as f32;
+                    let bm25 = bm25.entry(row_id).or_insert(0.0);
+                    *bm25 += self.idf(row_freq.len()) * freq * (K1 + 1.0)
+                        / (freq + K1 * (1.0 - B + B * self.docs.num_tokens(row_id) as f32 / avgdl));
```

Comment:
Is this formula correct? I would think `- B + B` is zero, so this looks suspicious. What is this based on? (perhaps you should have a link to this in the method?)
```suggestion
                        / (freq + K1 * (1.0 * self.docs.num_tokens(row_id) as f32 / avgdl));
```

Reply:
sure, I will add reference link for this method.
it's correct, it's `1.0 - B + (B * nq as f32 / avgdl)` but the operator `*` is with higher priority so I just ignored the parentheses